src/python/pants/backend/project_info/peek.py

```diff
@@ -161,22 +192,76 @@ def describe_ruleset(ruleset: DependencyRuleSet | None) -> tuple[str, ...] | Non
         return None
     return ruleset.peek()
 
+async def _create_target_alias_to_goals_map() -> dict[str, tuple[str, ...]]:
+    """
+    Returns a mapping from a target alias to the goals that can operate on that target.
+
+    For instance, `pex_binary` would map to `("run", "package")`.
+
+    :return: A mapping from a target alias to the goals that can operate on that target.
+    """
+    # TODO: This is manually curated for now - do we want to show all possible goals?
+    peekable_field_sets: list[type[FieldSet]] = [DeployFieldSet, PackageFieldSet, PublishFieldSet, RunFieldSet, TestFieldSet]
+    # TODO: How do we get the goal subsystem from field set? This shouldn't need to be manually maintained
+    field_set_to_goal_map: dict[type[FieldSet], str] = {
+        DeployFieldSet: "experimental-deploy",
+        PackageFieldSet: "package",
+        PublishFieldSet: "publish",
+        RunFieldSet: "run", 
+        TestFieldSet: "test",
+    }
+    assert len(peekable_field_sets) == len(field_set_to_goal_map), "Must have a goal string for each field set"
+    peekable_goals = [field_set_to_goal_map[fs] for fs in peekable_field_sets]
+
+    target_roots_to_field_sets_get = [Get(
```

Comment:
@stuhood @kaos  Any idea why this is only accessible when called from an `@goal_rule` rather than a regular `@rule`? 

Also, any idea why this would be returning nothing when run in unit tests using `run_pants` or `RuleRunner`? 

This is turning out to be surprisingly hard to write automated tests for.